### PR TITLE
Microsoft.AAD/DomainService: Add 2020-01-01 api-version

### PR DESCRIPTION
**Pull request by jihochang**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jihochang** [comment]
@fengzhou-msft Could you add the WaitForARMFeedback tag, so that the ARM team can approve this?

Also, I am not sure why the 3 automation tests are failing. Is the ignore test expected to fail? Any ideas?

**Comment by jihochang** [comment]
@fengzhou-msft any update??

**Comment by fengzhou-msft** [comment]
PrettierCheck is optional.

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json [hunk 1]
```diff
@@ -0,0 +1,805 @@
...
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "description": "The Resource model definition.",
+      "properties": {
+        "id": {
```
Comment (KrisBash) [inline specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: it's not required, but optimal to reference these from [common-types](https://github.com/Azure/azure-rest-api-specs/blob/a6f41b54a296e11631a65bbe4a2da623afb758a2/specification/common-types/resource-management/v1/types.json#L45)

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json [hunk 2]
```diff
@@ -0,0 +1,805 @@
...
+          },
+          {
+            "$ref": "#/parameters/DomainServiceNameParameter"
+          },
+          {
+            "name": "domainService",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DomainService"
```
Comment (KrisBash) [inline specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: Are all mutable properties in this request schema supported for PATCH?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json [hunk 3]
```diff
@@ -0,0 +1,805 @@
+{
```
[the remark this answers is not in the document]
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: Noted

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json [hunk 4]
```diff
@@ -0,0 +1,805 @@
...
+              "$ref": "#/definitions/DomainService"
+            }
+          },
+          "201": {
+            "description": "HTTP 201 (Created) if the operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/DomainService"
+            }
+          },
+          "202": {
```
[the remark this answers is not in the document]
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: they are both used

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json [hunk 5]
```diff
@@ -0,0 +1,805 @@
...
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
[the remark this answers is not in the document]
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: added defaults

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json (hunk 2 above)
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: they are supported using patch as a modify domain service

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json (hunk 1 above)
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json]: we will leave this alone for now

**Comment by jihochang** [comment]
@KrisBash - I updated the PR with your suggestions. Could you take another look? Thanks!

**Comment by jihochang** [comment]
@KrisBash - Could you take another look?

**Review by pilor — APPROVED** [review]

**Comment by jihochang** [comment]
@fengzhou-msft Could you merge this PR? ARM has approved the changes.

**Review by fengzhou-msft — APPROVED** [review]

**Comment by fengzhou-msft** [comment]
@jihochang can you mark the requested changes as addressed? I'm not able to merge the PR with it open.

**Comment by jihochang** [comment]
@fengzhou-msft I think Kristopher Bash needs to approve or be removed from the PR.

Can you try removing him and merging? I pinged him offline and he hasn't gotten back to me.

@KrisBash - could you approve this pull request?

**Review by KrisBash — APPROVED** [review]

**Pull request merged**